src/services/wido.ts

```diff
@@ -0,0 +1,197 @@
+import { getAddress } from "@ethersproject/address";
+import { TransactionRequest } from "@ethersproject/providers";
+import { approveForZap, getBalances, getSupportedTokens, getTokenAllowance, quote } from "wido";
+
+import { Chains, NETWORK_SETTINGS } from "../chain";
+import { Service } from "../common";
+import { usdc } from "../helpers";
+import { Address, Balance, Integer, Token, TokenAllowance } from "../types";
+
+export class WidoService extends Service {
+  async supportedTokens(): Promise<Token[]> {
+    // unsupported networks
+    if (this.chainId === 1337) {
+      throw new Error("Unsupported");
+    }
+    const networkSettings = NETWORK_SETTINGS[this.chainId];
+    const network = Chains[this.chainId];
+    const tokenList = await getSupportedTokens({
+      chainId: [this.chainId],
+      includeMetadata: true,
+      includePricing: true,
+      includeUnknown: false,
+    });
+
+    return tokenList.map((token) => {
+      return {
+        address: getAddress(token.address),
+        decimals: String(token.decimals),
+        icon: `https://assets.yearn.network/tokens/${network}/${token.address.toLowerCase()}.png`,
+        name: token.symbol,
+        priceUsdc: token.usdPrice,
+        dataSource: "wido",
+        supported: {
+          widoZapIn: true,
+          widoZapOut: networkSettings.zapOutTokenSymbols?.includes(token.symbol.toUpperCase()),
+        },
+        symbol: token.symbol,
+      };
+    });
+  }
+
+  async balances<T extends Address>(address: T): Promise<Balance[]> {
+    // unsupported networks
+    if (this.chainId === 1337) {
+      throw new Error("Unsupported");
+    }
+    const balances = await getBalances(address, [this.chainId]);
+
+    return balances.map((balance) => {
+      return {
+        address: getAddress(balance.address),
+        token: {
+          address,
```

Comment:
```suggestion
          address: getAddress(balance.address),
```
Both addresses should be of the token we are fetching the balanceOf